src/cmd/sign.js

```diff
@@ -125,6 +141,103 @@ export default function sign(
+) {
+  let oldUpdateManifest;
+  let oldUpdateManifestData: Object;
+  let statusCode;
+  let newUpdateManifest;
+  let updateManifestFileName = '';
+
+  if (manifestData !== undefined &&
+      manifestData.applications !== undefined &&
+      manifestData.applications.gecko !== undefined &&
+      manifestData.applications.gecko.update_url !== undefined) {
+    let parsed = url.parse(manifestData.applications.gecko.update_url);
+    if (parsed !== undefined && parsed.pathname !== undefined) {
+      updateManifestFileName = path.basename(parsed.pathname);
+    }
+
+    try {
+      [oldUpdateManifest, statusCode] = await httpFetchFile(
+        manifestData.applications.gecko.update_url);
+      if (statusCode < 200 || statusCode > 290) {
+        throw new WebExtError(
+          `Failed to retrieve ${updateManifestFileName}\n` +
+          `http statusCode error, server responded with: ${statusCode}`);
+      }
+    } catch (e) {
+      throw new WebExtError(
+        `Was unable to download ${updateManifestFileName}\nerror is: ${e}`);
+    }
+
+    try {
+      oldUpdateManifestData = JSON.parse(oldUpdateManifest);
+    } catch (e) {
+      throw new WebExtError(
+        `Unable to parse ${updateManifestFileName} file located at ` +
+        `${manifestData.applications.gecko.update_url}`);
+    }
+
+    if (updateLink.indexOf('{xpiFileName}') === -1) {
+      throw new WebExtError(
+        'Unable to parse --update-link url, please use {xpiFileName} ' +
+        'as a substitute for the XPI file');
+    }
+
+    let newVersion = {
+      version: manifestData.version,
+      update_link: updateLink.replace('{xpiFileName}', path.basename(XPIPath)),
+    };
+
+    if (oldUpdateManifestData) {
+      if (oldUpdateManifestData.addons === undefined) {
+        oldUpdateManifestData.addons = {};
+      }
+
+      if (id && oldUpdateManifestData.addons[id]) {
+        // append a new release to the updateManifest
+        if (oldUpdateManifestData.addons[id].updates) {
+          oldUpdateManifestData.addons[id].updates.push(newVersion);
+        }
+      } else {
+        // first release of this addon (or maybe the extension-id was botched? be careful!)
+        let addonName = manifestData.name || 'your application';
+        log.warn(`Creating first release of ${addonName}`);
+        log.warn(
+          'If this is not the actual first release, check your extension-id');
+        oldUpdateManifestData.addons[id] = {};
+        oldUpdateManifestData.addons[id].updates = [newVersion];
+      }
+
+      // the oldUpdateManifest has been updated, it's ready to be saved
+      newUpdateManifest = JSON.stringify(oldUpdateManifestData, null, 4);
+    } else {
+      throw new WebExtError('The .addons property is missing in the ' +
+        'retrieved updateManifest.json file');
+    }
+
+    return new Promise(function(resolve, reject) {
+      log.info('Saving to', path.join(artifactsDir, updateManifestFileName));
+      fs.writeFile(
```

Comment:
you can just `await fs.writeFile()` (it returns a promise since we're using [mz/fs](https://github.com/normalize/mz))